### PR TITLE
[Account] account alias create: Add --reseller-id

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Resolve https://github.com/Azure/azure-cli/issues/15436

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/account/azext_account/tests/latest/recordings/test_account.yaml [hunk 1]
```diff
@@ -12,18 +13,18 @@ interactions:
...
+      - --name --billing-scope --display-name --workload --reseller-id
       User-Agent:
-      - AZURECLI/2.11.1 azsdk-python-subscriptionclient/unknown Python/3.6.8 (Windows-10-10.0.19041-SP0)
+      - AZURECLI/2.13.0 azsdk-python-subscriptionclient/unknown Python/3.6.8 (Windows-10-10.0.19041-SP0)
     method: PUT
     uri: https://management.azure.com/providers/Microsoft.Subscription/aliases/cli_alias000001?api-version=2020-09-01
   response:
     body:
-      string: '{"id":"/providers/Microsoft.Subscription/aliases/cli_alias000001","name":"cli_alias000001","type":"Microsoft.Subscription/aliases","properties":{"subscriptionId":"00000000-0000-0000-0000-000000000000","provisioningState":"Accepted"}}'
+      string: '{"id":"/providers/Microsoft.Subscription/aliases/cli_alias000001","name":"cli_alias000001","type":"Microsoft.Subscription/aliases","properties":{"subscriptionId":"89b84889-2fa2-40f4-a7c1-f2f0de19ac39","provisioningState":"Accepted"}}'
```
Comment (yungezz) [inline src/account/azext_account/tests/latest/recordings/test_account.yaml:27]: why this subs id not masked?

File: src/account/azext_account/vendored_sdks/subscription/models/_models.py [hunk 2]
```diff
@@ -316,24 +316,28 @@ class PutAliasRequestProperties(msrest.serialization.Model):
...
 
     def __init__(
         self,
         **kwargs
     ):
         super(PutAliasRequestProperties, self).__init__(**kwargs)
-        self.display_name = kwargs['display_name']
-        self.workload = kwargs['workload']
-        self.billing_scope = kwargs['billing_scope']
+        self.display_name = kwargs.get('display_name', None)
```
Comment (yungezz) [inline src/account/azext_account/vendored_sdks/subscription/models/_models.py:336]: this looks like manual fixing :) right? will it be overwritten in next generation?

**Review by yungezz — APPROVED** [review]

File: src/account/azext_account/vendored_sdks/subscription/models/_models.py (hunk 2 above)
Comment (fengzhou-msft) [reply src/account/azext_account/vendored_sdks/subscription/models/_models.py:336]: The change is auto-generated and caused by a fix in the swagger to mark the property as optional.

File: src/account/azext_account/tests/latest/recordings/test_account.yaml (hunk 1 above)
Comment (fengzhou-msft) [reply src/account/azext_account/tests/latest/recordings/test_account.yaml:27]: subscription id is only masked in limited cases by [azure_devtools](https://github.com/Azure/azure-python-devtools/blob/4be812e0237572afa23c37c259c414be79557e54/src/azure_devtools/scenario_tests/recording_processors.py#L50-L69). I may have manually masked it last time.

**Review by arrownj — APPROVED** [review]

**Pull request merged**